esmvalcore/preprocessor/_supplementary_vars.py

```diff
@@ -103,6 +104,123 @@ def add_cell_measure(
     )
 
 
+def get_data_dims(  # noqa: C901
+    cube_c: Cube,
+    cube_a: Cube | iris.coords.AncillaryVariable,
+    cube_a_var: iris.coords.AncillaryVariable,
+) -> list[None | int]:
+    """Get matching data dimensions between cube and ancillary variable.
+
+    Parameters
+    ----------
+    cube_c: Cube
+        Iris cube with input data.
+    cube_a: Cube or iris.coords.AncillaryVariable
+        Iris cube or AncillaryVariable with ancillary data.
+    cube_a_var: iris.coords.AncillaryVariable
+        Ancillary variable object containing the ancillary data.
+
+    Returns
+    -------
+    list
+        List with as many entries as the ancillary variable dimensions.
+        The i-th entry corresponds to the i-th ancillary variable
+        dimension match with the cube's dimensions. If there is no match
+        between the ancillary variable and the cube for a dimension, then entry
+        defaults to None.
+    """
+    # Match the coordinates of the ancillary cube to coordinates and
+    # dimensions in the input cube before adding the ancillary variable.
+    data_dims: list[None | int] = []
+    if isinstance(cube_a, iris.coords.AncillaryVariable):
+        start_dim = cube_c.ndim - len(cube_a_var.shape)
+        data_dims = list(range(start_dim, cube_c.ndim))
+    else:
+        data_dims = [None] * cube_a.ndim
+        for coord in cube_a.coords():
+            logger.debug("Matching coordinate...\n%s", coord)
+            try:
+                for ancillary_dim, cube_dim in zip(
+                    cube_a.coord_dims(coord),
+                    cube_c.coord_dims(coord),
+                    strict=False,
+                ):
+                    data_dims[ancillary_dim] = cube_dim
+                logger.debug("Matched coordinate.")
+            except iris.exceptions.CoordinateNotFoundError:
+                logger.debug(
+                    "No metadata match from ancillary coord in cube coords.",
+                )
+                cube_dims = []
```

Comment:
```suggestion
                cube_dims = None
```
Should the default value be `None` here, to signal that no coordinate was found on line 209?